### PR TITLE
Creating a Fleet creates a GameServerSet

### PR DESCRIPTION
**Pull request by markmandel**
All this commit is doing is create a GameServerSet when a Fleet is created. Also, if the Replicas in
a Fleet are updated, then the owned GameServerSet's Replicas are also updated.

For this commit, we are not worrying about handling updates to the Spec (e.g. new image) to the Fleet,
and doing updates to a live Fleet, we are just concerned with scaling up and down.

Parent ticket: #70

### CODE REVIEW

**Comment by fooock** [comment]
@markmandel can be the error in this [line](https://github.com/GoogleCloudPlatform/agones/blob/master/build/Makefile#L110)? `-mkdir` instead `mkdir`

**Comment by markmandel** [comment]
@fooock - the `-` just says to `Make`, if this command fails, just ignore it.
Since the directory could already exist from previous runs, we just ignore the `mkdir` command

The test failed because I didn't run `gen-install` to push the changes I made to the Helm chart to the manual install. `diff` actually returns a non-zero exit code when it finds a difference :smile:

**Comment by fooock** [comment]
ahhh ok, thanks for the explaination @markmandel . I didn't know it 😃

File: pkg/fleets/controller.go [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+		activeGsSet, err = c.gameServerSetGetter.GameServerSets(fleet.ObjectMeta.Namespace).Create(activeGsSet)
+		if err != nil {
+			return errors.Wrapf(err, "error creating gameserverset for fleet %s", fleet.ObjectMeta.Name)
+		}
+
+		c.recorder.Eventf(fleet, corev1.EventTypeNormal, "CreatingGameServerSet",
+			"Created GameServerSet %s", activeGsSet.ObjectMeta.Name)
+
+	} else {
+		// for now, we're ignoring any change to the template - will handle on the next PR
```
Comment (EricFortin) [inline pkg/fleets/controller.go]: You want to support multiple game server set to implement things like rolling update?

File: pkg/fleets/controller.go [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+	// if the replica count has changed, then update the GameServerSet
+	if fleet.Spec.Replicas != activeGsSet.Spec.Replicas {
+		gsSetCopy := activeGsSet.DeepCopy()
+		gsSetCopy.Spec.Replicas = fleet.Spec.Replicas
+
+		if gsSetCopy, err = c.gameServerSetGetter.GameServerSets(fleet.ObjectMeta.Namespace).Update(gsSetCopy); err != nil {
+			return errors.Wrapf(err, "error updating replicas for gameserverset for fleet %s", fleet.ObjectMeta.Name)
+		}
+		c.recorder.Eventf(fleet, corev1.EventTypeNormal, "ScalingGameServerSet",
+			"Scaling GameServerSet %s to %d", gsSetCopy.ObjectMeta.Name, gsSetCopy.Spec.Replicas)
```
Comment (EricFortin) [inline pkg/fleets/controller.go]: Could we add the old number to in the event so we know if it was scaling up or down?

File: pkg/fleets/controller.go (hunk 1 above)
Comment (markmandel) [reply pkg/fleets/controller.go]: Yep, exactly!

File: pkg/fleets/controller.go (hunk 2 above)
Comment (markmandel) [reply pkg/fleets/controller.go]: Nice! Done.

**Pull request merged**